internal/util/logging.go

```diff
@@ -4,75 +4,105 @@ import (
 	"fmt"
 	"io"
 
-	"github.com/sirupsen/logrus"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 )
 
-// we currently implement two different loggers and use a middleware called
-// logrusr to translate logrus loggers into go-logrs (used by controller-runtime).
-// The middleware currently squashes loglevels 0-4 together and effectively starts
-// the "info" level logging at 0 (whereas logrus starts that at 4 (range from 0)).
-// Since the middleware at the time of writing made all of this part of the private
-// implementation these options are used for convenience until a time when we unify
-// our logging implementations into one or the other.
-//
-// See: https://github.com/Kong/kubernetes-ingress-controller/issues/1893
+// logr does not use semantic log types. Logs are either informational or error messages with arbitrary verbosity,
```

Comment:
I'd suggest moving it out of the package `utils` to some log-dedicated package with a better name e.g. `konglog` or something like that

Reply:
Maybe in this package helpers like `NewNoOpLogger()` could be introduced and used across the codebase rather than explicit 
```go
zapr.NewLogger(zap.NewNop())
```
in many places, it encapsulates `zapr` and `zap` making them implementation detail. Maybe they should be only imported in this package... Something to consider

Reply:
The places where you set up a logger aren't implementation agnostic. logr defines how you log calls, but not how you manipulate the underlying logger. This mostly matters for tests (the application code should use the same base logger created by `SetupLoggers()` throughout), and the way zap and logrus handle log observation doesn't follow any standard interface or immediately lend itself to one.

If the setup code required is short, I do tend to just copy/paste it, since I'd rather that than try to expand a helper when I hit some future test that has breaking needs the original helper didn't account for.